### PR TITLE
Prepare a doc to help ebable autofdo for nebulagraph

### PR DESCRIPTION
**Pull request by lipzhu**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
https://github.com/vesoft-inc/nebula/issues/4636
## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by lipzhu** [comment]
@wey-gu Do you want to build a process to generate a profile regularly and provide an option for users to enable the AutoFDO when building from source?

**Comment by wey-gu** [comment]
Oops, Sorry @lipzhu I missed this mention for a month, shame for this.

Thanks @xtcyclist @yixinglu btw.

**Comment by lipzhu** [comment]
@wey-gu Its my apologies that delay the document so long. Hope you still have interests about this :)

File: docs/autofdo/enable_autofdo_for_nebulagraph.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+## Recompile GraphNebula Binary with the Merged Profile
+
+Recompile the GraphNebula Binary by passing the profile with compile option `-fauto-profile`
+```diff
+diff --git a/cmake/nebula/GeneralCompilerConfig.cmake b/cmake/nebula/GeneralCompilerConfig.cmake
+@@ -20,6 +20,8 @@ add_compile_options(-Wshadow)
+ add_compile_options(-Wnon-virtual-dtor)
+ add_compile_options(-Woverloaded-virtual)
+ add_compile_options(-Wignored-qualifiers)
++add_compile_options(-fauto-profile=~/fbdata.afdo)
```
[the remark this answers is not in the document]
Comment (lipzhu) [reply docs/autofdo/enable_autofdo_for_nebulagraph.md:112]: Agree. We can add an default option if the user want to use the default profile provided by community. 
And also provide an option to  the users who want to build their own profile data based on their real production environment.

**Comment by lipzhu** [comment]
> Thanks for this experiment and contribution.
> 
> I wonder if there are ways to automate the autofdo build process. After all, we have to integrate it to our standard release workflow.

My suggestion is to integrate the work into the community standard build workflow, the community refreshed the profile periodically. 
Just as chromium community did https://chromium.googlesource.com/chromiumos/chromite/+/master/cbuildbot/afdo.py.

**Comment by wey-gu** [comment]
> It takes a lot of effort to integrate this to our build process. But I think the effort is worthy, we will consider to utilize it seriously in the develop plan.

Wow, looking forward to it!

> Before that, I sugguest we could add this doc to our official docment as an alternative build method, instead of putting it in our source tree.

Agreed, this will enable those who would like to benefit from autofdo before we bring it to the build process.

@whitewum what do you think, please? could we somehow put the guide of this PR to docs in a proper chapter and co-author @lipzhu for the docs PR?

**Comment by whitewum** [comment]
> > It takes a lot of effort to integrate this to our build process. But I think the effort is worthy, we will consider to utilize it seriously in the develop plan.
> 
> Wow, looking forward to it!
> 
> > Before that, I sugguest we could add this doc to our official docment as an alternative build method, instead of putting it in our source tree.
> 
> Agreed, this will enable those who would like to benefit from autofdo before we bring it to the build process.
> 
> @whitewum what do you think, please? could we somehow put the guide of this PR to docs in a proper chapter and co-author @lipzhu for the docs PR?

Sure. 
Great work.

**Comment by lipzhu** [comment]
> It takes a lot of effort to integrate this to our build process. But I think the effort is worthy, we will consider to utilize it seriously in the develop plan. Before that, I sugguest we could add this doc to our official docment as an alternative build method, instead of putting it in our source tree.

Cool~ looking forward to the official release version of NebulaGraph with AFDO.

**Comment by wey-gu** [comment]
@dutor I would like to suggest merging this file in the core repo as is(or in some other path), now, what do you think, please?

**Comment by lipzhu** [comment]
> > 
> 
> Sorry about the late reply.
> 
> It takes a lot of effort to integrate this to our build process. But I think the effort is worthy, we will consider to utilize it seriously in the develop plan. Before that, I sugguest we could add this doc to our official docment as an alternative build method, instead of putting it in our source tree.
> 
> @wey-gu What do you think?

BTW, do you mind sharing the performance data about how afdo can gain from your tests?

**Pull request closed without merging**